### DISCUSSION TITLE
Button is always disabled when i add a Command to it

### DISCUSSION THREAD

**Discussion by Avngarde** [AvaloniaUI/Avalonia#6856] [Q&A] [answered]
Hello!

I don't know what to do with this problem so i ask it here. When i add a Command to my Button, it's always disabled. Here is the XAML code of my button:

`<Button Grid.Row="1" Background="#0078d4" FontSize="20" HorizontalAlignment="Center" Command="{Binding AddNewEvent}" Content="Test" Margin="0,70,0,0" Foreground="White">Add new</Button>`

And here is the method code in ViewModel:

`        public void AddNewEvent()
        {
            Console.WriteLine("Test text");
        }`

**Comment by workgroupengineering** [comment]
https://docs.avaloniaui.net/docs/data-binding/binding-to-commands#binding-to-methods

**Comment by Avngarde** [comment, marked as answer]
Nevermind, i fixed it.

I was making new Window appear like this:

```csharp
            Window addNewWindow = new AddNewWindow();
            addNewWindow.Show();
````

But then i added DataContext to it

```csharp
            Window addNewWindow = new AddNewWindow();
            addNewWindow.DataContext = new AddNewWindowViewModel();
            addNewWindow.Show();
```

And everything worked nice.
